<launch>
  <!-- Arguments -->
  <arg name="rviz" default="true"/>
  <arg name="pclviz" default="false"/>

  <arg name="registeredCloud" default="/velodyne_cloud_registered"/>
  <arg name="velodyneCloud" default="/velodyne_points"/>

  <arg name="registered" default="10"/>
  <arg name="velodyne" default="1"/>

  <!-- Parameters -->
  <param name="windowSize" value="10" type="int"/>
  <param name="boxFilterBoundXYZ" value="5.0 5.0 5.0" type="string"/>
  <param name="voxelGridLeafSize" value="0.5" type="double"/>
  <param name="neighborRadius" value="0.5" type="double"/>
  <param name="weightingFactor" value=".2" type="double"/>  

  <param name="normalDistWeight" value="0.1" type="double"/>
  <param name="maxIterations" value="10000" type="int"/>
  <param name="distThreshold" value="0.05" type="double"/>
  <param name="radiusLimits" value="0 15" type="string"/>

  <!--RVIZ params-->  
  <param name="displayCloud" value="true" type="bool"/>
  <param name="displayNormals" value="true" type="bool"/>
  <param name="displayCenterAxis" value="true" type="bool"/>
  <param name="displayCylinder" value="false" type="bool"/>

  <param name="usePCLViz" value="$(arg pclviz)" type="bool"/>

	<!-- Publish Point Cloud From Bag -->
	<node pkg="geometric_mapping" type="geometric_mapping_node" name="geometric_mapping_node" output="screen">
    <!-- Remapping -->
    <remap from="inputCloud" to="$(arg registeredCloud)"/>
    <remap from="inputOdometry" to=""/>

    <remap from="cloudOutput" to="/choppedCloud"/>
    <remap from="normalsOutput" to="/surfaceNormals"/>
    <remap from="eigenBasisOutput" to="/eigenBasis"/>
    <remap from="cylinderOutput" to="/cylinderModel"/>
  </node>

  	<!-- Play Bag -->
<!-- 	<arg name="path" default="-d $(find pcl_practice)../../../mine_bags/"/>
	<arg name="file" default="mine_data_2018-05-14-14-42-01_54"/>
	<node pkg="rosbag" type="play" name="player" output="screen" args="-l $(arg path)$(arg file).bag"/> -->

  <!-- Visualize the point cloud in rviz -->
  <group if="$(arg rviz)">
    <node launch-prefix="nice" name="rviz" pkg="rviz" type="rviz" args="-d $(find geometric_mapping)/rviz_config/mapping.rviz" required="true"/>
  </group>
</launch>